melodies_monet/plots/ozone_sonder_plots.py

```diff
@@ -0,0 +1,230 @@
+            #add a line plot
+            ax.plot(O3_MODEL_ALL[i],ALT_gl,'-*',color=fill_color_list[i+1],label = 'MODEL:'+model_name_list[i+1])
+            
+            colors=[fill_color_list[i+1]]*(len(altitude_threshold_list)-1)
+            for patch,color in zip(bplot_model['boxes'],colors):
+                patch.set_edgecolor(color)
+
+        plt.title('Comparison at '+str(station_name[0])+' on '+str(release_time)+' UTC')
+        plt.legend()
+        plt.ylim(altitude_range[0],altitude_range[1])
+        plt.xlim(ozone_range[0],ozone_range[1])
+        plt.ylabel('ALT-ground level (km)')
+        plt.xlabel('O3 (ppbv)')
+
+
+def split_by_threshold(o3_list_input,alt_list_input,threshold_list_input):
+    df = pd.DataFrame(data={'o3':o3_list_input,'alt':alt_list_input})
+    output_list = []
+    for i in range(1,len(threshold_list_input)):
+        df_here = df.o3.loc[(df.alt>threshold_list_input[i-1])&(df.alt<=threshold_list_input[i])]
+        output_list.append(df_here.values)
+    return output_list
+
+
+def density_scatter_plot_os(df,altitude_range,ozone_range,station_name,altitude_method,cmap_method):
+    #release height info,get height of each release site to be substract    
+    df_height = pd.DataFrame({
+         'station':['Boulder, Colorado','Huntsville, Alabama','University of Rhode Island','Trinidad Head, California'],
+         'height':[1.743,0.203,0.021,0.046]
+         })
+    if altitude_method[0] == 'ground level':
+        height_value = df_height.loc[df_height['station']==station_name[0]].values[0][1]
+    elif altitude_method[0] == 'sea level':
+        height_value = 0
+
+    #get o3 model, o3 sonder (obs) and height
+    df_short = df[df['altitude']<altitude_range[1]+height_value]
+    ALT = df_short['altitude']-height_value
+    O3_OBS = df_short['o3']
+    O3_MODEL = df_short['o3_ave']
+
+    #plot scatter and colorbar
+    sc=plt.scatter(O3_OBS,O3_MODEL,c= ALT,vmin=altitude_range[0],vmax=altitude_range[1],cmap = 'jet',edgecolors='k',linewidth=0.5,s = 15)
+    cb=plt.colorbar(sc)
+    if altitude_method[0] == 'ground level':
+        cb.set_label('Ground Level Altitude (km)',fontsize=15)
+    elif altitude_method[0] == 'sea level':
+        cb.set_label('Sea Level Altitude (km)',fontsize=15)
+
+    #add some points to make best fit go entire domain
+    slope = np.poly1d(np.polyfit(O3_OBS, O3_MODEL, 1))
+    Modify_OBS = O3_OBS.to_list()
+    Modify_OBS.append(ozone_range[0])
+    Modify_OBS.append(ozone_range[1])
+    Modify_OBS.sort()
+    plt.plot(Modify_OBS, slope(Modify_OBS),color='k',linestyle='-.',label='best fit')
+   
+    #plot Y=X line
+    plt.axline((ozone_range[1],ozone_range[1]),slope=1,color='k',linestyle='-',label='Y=X')
+
+    plt.xlim(ozone_range[0],ozone_range[1])
+    plt.ylim(ozone_range[0],ozone_range[1])
+    plt.xlabel('O$_3$ Obs (ppbv)')
+    plt.ylabel('O$_3$ Model (ppbv)')
+    plt.legend()
```

Comment:
Once you code the vmin_plot and vmax_plot in you will also be able to access the ylabel_plot and could use that here too to make this code more universal and not just for Ozone. And you could expand your code to look at temperature for example very easily with some small updates like this.

You can just use the text in ylabel_plot and then append Obs to the xlabel and append Model to the ylabel here to make this more generic.

Reply:
this is done!